jsonschema-hyperschema.xml

```diff
@@ -703,9 +704,26 @@ GET /foo/
 
             <section title="targetSchema">
                 <t>
-                    This property provides a schema that is expected to describe the link target, including what a client can expect if it makes an HTTP GET request, and what it should send if it replaces the resource in an HTTP PUT request. This property is advisory only.
+                    This property provides a schema that is expected to describe
+                    the link target's representation.  Depending on the protocol,
+                    the schema may or may not describe the response to any particular
+                    request sent to the link.  This property is advisory only.
                 </t>
-
+                <section title="&quot;targetSchema&quot; and HTTP">
+                    <t>
+                        The relationship between a resource's representation and
+                        HTTP requests and responses is determined by
+                        <xref target="RFC7231">RFC 7231, section 4.3.1 - "GET", section 4.3.4 "PUT", and section 3.1.4.2, "Content-Location"</xref>.
+                        In particular, "targetSchema" suggests what a client can expect
```

Comment:
We did adjust the definition of `"readOnly"` to clarify that it means "under the control of the server" and not "you are forbidden from including this field in PUTs" (the latter would force violation of HTTP semantics so that would be bad).

The server can decide whether it needs to detect changes to such fields and raise errors (400 or 409), or whether to silently ignore things.  For instance a field that is present to reflect state changes elsewhere probably should never raise an error because of the following scenario:

* GET on /foo/1, field "x" has value "y"
* some request on /bar/9 causes /foo/1's "x" to change value to "z"
* PUT the representation back to /foo/1 modifying the value of field "w"

On that last put, "x" will still have value "y" but on the server, its value is now "z".  So that could look like a change.  But as a "readOnly" field, the server is allowed to ignore that apparent change.

A resource can also offer different levels of enforcement using [the "Prefer" header](https://tools.ietf.org/html/rfc7240#section-4.4) value's "handling=strict" or "handling=lenient" options.

The above x/y/z/w example probably shouldn't cause an error either way, but for a field that is not likely to change out from under you during a GET/modify/PUT, in lenient mode you could silently ignore changes, while returning a 409 or 400 in strict mode.